### PR TITLE
Update dotstrings parser to handle entries with multi-line values

### PR DESCRIPTION
**Pull request by fla-t**
The dotstrings parser has been updated to handle multi-line entries in the format 
```
"key" = "value
that spans
multiple lines"; 
```
This allows for more flexibility in writing and parsing dotstrings files

### CODE REVIEW

File: dotstrings/parser.py [hunk 1]
```diff
@@ -14,7 +14,7 @@ class Patterns:
 
     comment = re.compile(r"(\'(?:[^\'\\]|\\[\s\S])*\')|//.*|/\*(?:[^*]|\*(?!/))*\*/", re.MULTILINE)
     whitespace = re.compile(r"\s*", re.MULTILINE)
-    entry = re.compile(r'"(.*)"\s*=\s*"(.*)";')
+    entry = re.compile(r'"([^"]*?)"\s*=\s*"((?:[^";]|"(?!\s*;))*?)";', re.DOTALL)
```
Comment (dalemyers) [inline dotstrings/parser.py:17]: Unfortunately this will break on strings that contain quotes. e.g. `NSLocalizedString("Hello \"World\"", "Some Comment")` gets turned into `"Hello \"World\"" = "Hello \"World\"";`. 

I think the second change on this line will also stop the string containing `;`.

File: tests/test_simple.py [hunk 2]
```diff
@@ -2,12 +2,12 @@
 
 # pylint: disable=line-too-long
 
+import dotstrings
 import os
 import sys
 import unittest
 
 sys.path.insert(0, os.path.abspath(os.path.join(os.path.abspath(__file__), "..", "..")))
-import dotstrings
```
Comment (dalemyers) [inline tests/test_simple.py]: This needs to be below the `sys.path.insert` or it won't work if you haven't "installed" the package.

File: tests/test_simple.py [hunk 3]
```diff
@@ -86,9 +86,10 @@ def test_strings_format(self):
         entries = dotstrings.load(strings_file_path)
 
         self.assertEqual(entries[0].strings_format(), '/* This is a\n   multiline comment */\n"carry" = "breathe";')
-        self.assertEqual(entries[1].strings_format(), '/* This is another\n   multiline comment, but for a dupe */\n"condition" = "outgoing";')
+        self.assertEqual(entries[1].strings_format(
+        ), '/* This is another\n   multiline comment, but for a dupe */\n"condition" = "outgoing";')
```
Comment (dalemyers) [inline tests/test_simple.py]: Nit: Odd formatting here.

**Review by dalemyers — CHANGES_REQUESTED** [review]

File: dotstrings/parser.py (hunk 1 above)
Comment (fla-t) [reply dotstrings/parser.py:17]: The first change is there so that the first match group doesn't end up matching the whole string (even matching the "value" part).

But yeah it wouldn't work when there are more than two quotes, in the "key" part of the string (the two quotes around the key itself). This was intentional because I didn't thought there would be any "key" that would have more quotes than two.

If we really want to cater keys that have quotes inside of them, we can replicate the regex that we have for value part, but without the semicolon

```
entry = re.compile(r'"((?:[^";]|"(?!\s*;))*?)"\s*=\s*"((?:[^";]|"(?!\s*;))*?)";')
```

Comment (fla-t) [reply dotstrings/parser.py:17]: The second change is finding a ";" to match the value part of the string.

Comment (fla-t) [reply dotstrings/parser.py:17]: Can you change the test file I have here to contain edge cases?

**Comment by fla-t** [comment]
Also there is a line that I hate everytime I take a look at it. 

[comment = comment[::-1].replace("/*", "", 1)[::-1]](https://github.com/fla-t/dotstrings/blob/84ff6d3d079e37d09b56548ce754d11d48adfe21/dotstrings/parser.py#L129)

```
# can easily be this
comment = comment.removeprefix("/*")
comment = comment.removesuffix("*/")
```

Or maybe I am missing something? :) 
(I would love if we can add this change in this PR)
